pkg/tfgen/generate.go

```diff
@@ -1546,9 +1560,12 @@ func resourceName(provider string, rawname string,
 	}
 	if info == nil || info.Tok == "" {
 		// default transformations.
-		name := withoutPackageName(provider, rawname)                  // strip off the pkg prefix.
-		camel := tfbridge.TerraformToPulumiName(name, nil, nil, false) // camelCase the module name.
-		pascal := tfbridge.TerraformToPulumiName(name, nil, nil, true) // PascalCase the resource name.
+		name := withoutPackageName(provider, rawname)              // strip off the pkg prefix.
+		camel := tfbridge.TerraformToPulumiNameV2(name, nil, nil)  // camelCase the module name.
+		pascal := tfbridge.TerraformToPulumiNameV2(name, nil, nil) // PascalCase the resource name.
+		if pascal != "" {
+			pascal = string(unicode.ToUpper(rune(pascal[0]))) + pascal[1:]
```

Comment:
What's this change doing here? This feels unexpected?

Reply:
Previously, the code looked like this:
https://github.com/pulumi/pulumi-terraform-bridge/blob/59a7281edaa87aa3e7c3d8544521d9d4232c05e3/pkg/tfgen/generate.go#L1550-L1551

Since `tfbrige.TerraformToPulumiName` doesn't support a (non-reversible) PascalCase option anymore, it needed to be manually reimplemented.

Reply:
Got it. That makes sense.